scripts/bump-executable-spec.sh

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+set -e
+
+EXEC_SPEC_REPO_DIR=$1
+if [ ! $EXEC_SPEC_REPO_DIR ]; then
+  echo "No repository path provided."
+  echo "Usage: ./bump-executable-spec.sh EXEC_SPEC_REPO_DIR"
+  exit 1
+fi
+
+AGDA_COMMIT=$(git rev-parse HEAD)
+echo "Building executable spec.."
+nix-build -A ledger.hsSrc
+EXEC_SPEC_OUT=$(readlink -f result/haskell/Ledger)
+
+echo "Entering executable spec repo.."
+chmod 755 -R $EXEC_SPEC_REPO_DIR
+pushd $EXEC_SPEC_REPO_DIR
+
+echo "Removing old files.."
+FILES_TO_RM=$(git ls-tree -r main --name-only | grep -vE "flake.lock|flake.nix|CHANGELOG.md|.gitignore")
+for f in $FILES_TO_RM; do
+  rm -rf $f
+done
```

Comment:
Is the purpose of this to not have anything but the Haskell code on the branch? Running `rm -rf` with a variable argument always makes me nervous.

I think the better way might be to have an empty branch from the beginning, then check that out and only add the new files.

Reply:
It's the opposite actually, the grep command is inverted.